src/control/server/ctl_smd_rpc.go

```diff
@@ -294,9 +294,6 @@ func (svc *ControlService) mapIDsToEngine(ctx context.Context, ids string, useTr
 			return nil, errors.Errorf("failed to retrieve instance for rank %d", rr.Rank)
 		}
 		eisPtr := &eis[0]
-
-		edm[eisPtr] = []devID{}
-
 		for _, dev := range rr.Devices {
```

Comment:
@knard-intel and I had a discussion this morning and I said I'd look again later. I'm looking at the code again now and I actually can't see why this makes any difference at all. There should be a one-to-one mapping between Rank and engine instance, so for each iteration of the loop, initialising the entry for the instance key's value should be a valid operation (although probably unnecessary). The only time this theoretically would cause an issue is if during the iteration over the response ranks, the same engine instance is returned from the `FilterInstanceByRankSet()` lookup which shouldn't happen in practice. I will keep looking... It might be worth adding a check to fail if there is already an entry in the map for the returned engine instance.

Reply:
OK, thanks for the explanations. I still don't understand why the empty entry was added in the first place, but if you're both confident that this won't have unintended consequences then I'll accept that it's a reasonable solution. I'm certainly guilty of forgetting to remove code that isn't necessary any more. It does seem like this area needs more test coverage, so it's good that you're planning to work on that as soon as possible.